core/capabilities/ccip/ccipevm/addresscodec.go

```diff
@@ -0,0 +1,24 @@
+package ccipevm
+
+import (
+	"fmt"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+)
+
+type AddressCodec struct{}
+
+func (a AddressCodec) AddressBytesToString(addr []byte) (string, error) {
+	if len(addr) != common.AddressLength {
+		return "", fmt.Errorf("invalid EVM address length, expected %v, got %d", common.AddressLength, len(addr))
+	}
+	return hexutil.Encode(addr), nil
```

Comment:
Instead of this, use this version:
common.BytesToAddress(addr).Hex()

The reason is because the latter is EIP55-compliant.
More context: https://medium.com/@zakhard/eip-55-explained-solving-the-address-checksum-problem-01ac2bb0efc4